src/design-system/components/Button/Button.tsx

```diff
@@ -5,25 +5,34 @@ import { forwardRef } from 'react';
 
 const Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
   if (props.htmlElement === 'a') {
+    // Remove htmlElement from props
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { htmlElement, ...rest } = props;
+
     return (
       <Link
         href={props.href}
         className={`button ${
           props.variant === 'secondary' ? `button--secondary` : ''
         }`}
-        {...props}
+        {...rest}
       >
         {props.children}
       </Link>
     );
   }
+
+  // Remove htmlElement from props
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { htmlElement, ...rest } = props;
```

Comment:
Maybe this would make the linter happier? Underscores are understood to be unused variables…

```suggestion
  const { htmlElement: _, ...rest } = props;
```